py/selenium/webdriver/firefox/service.py

```diff
@@ -41,12 +41,13 @@ def __init__(self, executable_path: str = DEFAULT_EXECUTABLE_PATH,
         :param service_args: Optional list of arguments to pass to the
             GeckoDriver binary.
         :param log_path: Optional path for the GeckoDriver to log to.
-            Defaults to _geckodriver.log_ in the current working directory.
+            Can also be called with (path_name, mode).
+            Defaults to _geckodriver.log_ in the current working directory and _a+_ respectively.
         :param env: Optional dictionary of output variables to expose
             in the services' environment.
-
         """
-        log_file = open(log_path, "a+") if log_path else None
+        log_path, log_path_mode = (log_path[0], log_path[1]) if isinstance(log_path,Tuple) else (log_path, "a+")
```

Comment:
How about:

```
open_args = (log_path,) if isinstance(log_path, str) else log_path
log_file = open(*open_args)
```
Has the added benefit of allowing the caller to shovel arguments into the `open` call positionally (should there be another use case to change something else and not just `mode`)

Reply:
Okay, I think I did what you asked, but I'm not sure I 100% I understood you.